test/fluent/plugin/out_splunk_hec_test.rb

```diff
@@ -243,6 +243,20 @@
     end
   end
 
+  describe 'timeout params' do
+    it 'should reset unused connection after 5 seconds' do
+      expect(create_output_driver('hec_host idle_timeout', 5).instance.idle_timeout).must_equal 5
+    end
+
+    it 'should allow nil between reading chunks from the socket' do
+      expect(create_output_driver('hec_host read_timeout', 5).instance.read_timeout).must_equal nil
```

Comment:
I think we need more test cases when we actually specify a timeout and see if that is reflected.

Reply:
@chaitanyaphalak 
How do I successfully override the NIL value set in the out_splunk_hec.rb file? I'm trying something like the following, unsuccessfully.
```
    it 'should allow a timeout value between reading chunks from the socket' do
      val = 10
      expect(create_output_driver("hec_host read_timeout#{val}", 5).instance.read_timeout).must_equal 10
    end
```
Pointers?

Reply:
You will have to pass a config with those not nil values :)